lib/matplotlib/figure.py

```diff
@@ -3185,24 +3214,21 @@ def tight_layout(self, *, pad=1.08, h_pad=None, w_pad=None, rect=None):
 
         See Also
         --------
-        .Figure.set_tight_layout
+        .Figure.set_layout_engine
         .pyplot.tight_layout
         """
-        from contextlib import nullcontext
-        from .tight_layout import (
-            get_subplotspec_list, get_tight_layout_figure)
+        from .tight_layout import get_subplotspec_list
         subplotspec_list = get_subplotspec_list(self.axes)
         if None in subplotspec_list:
             _api.warn_external("This figure includes Axes that are not "
                                "compatible with tight_layout, so results "
                                "might be incorrect.")
-        renderer = _get_renderer(self)
-        with getattr(renderer, "_draw_disabled", nullcontext)():
-            kwargs = get_tight_layout_figure(
-                self, self.axes, subplotspec_list, renderer,
-                pad=pad, h_pad=h_pad, w_pad=w_pad, rect=rect)
-        if kwargs:
-            self.subplots_adjust(**kwargs)
+        # note that here we do not _set_ the figures engine to tight_layout
+        # but rather just perform the layout in place for back compatibility.
+        engine = TightLayoutEngine(**kwargs)
+        self.set_layout_engine(engine)
+        engine.execute()
+        self.set_layout_engine(None)
```

Comment:
The other `execute` call says that it might raise `ValueError`, so resetting the layout engine should be done in a `finally`.

Reply:
I think the `try/finally` is the same?